delphi-frontend/src/test/java/au/com/integradev/delphi/utils/files/DelphiFileUtils.java

```diff
@@ -23,18 +23,34 @@
 
 import au.com.integradev.delphi.DelphiProperties;
 import au.com.integradev.delphi.compiler.Platform;
+import au.com.integradev.delphi.file.DelphiFile;
 import au.com.integradev.delphi.file.DelphiFileConfig;
 import au.com.integradev.delphi.preprocessor.DelphiPreprocessorFactory;
 import au.com.integradev.delphi.preprocessor.search.SearchPath;
 import au.com.integradev.delphi.utils.types.TypeFactoryUtils;
+import java.io.IOException;
+import java.io.UncheckedIOException;
 import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.Collections;
 
 public final class DelphiFileUtils {
   private DelphiFileUtils() {
     // Utility class
   }
 
+  public static DelphiFile parse(String content) {
+    try {
+      Path path = Files.createTempFile(null, ".pas");
+      Files.writeString(path, content, StandardCharsets.UTF_8);
```

Comment:
This should also write a UTF-8 BOM:

```suggestion
     Files.writeString(path, "\uFEFF" + content, StandardCharsets.UTF_8);
```

Reply:
Done.